mangaki/mangaki/views.py

```diff
@@ -749,6 +751,29 @@ def update_newsletter(request):
     return HttpResponse()
 
 
+def update_research(request):
+    is_ok = None
+    if request.user.is_authenticated and request.method == 'POST':  # Toggle on one's profile
+        username = request.user.username
+        is_ok = request.POST['research_ok'] == 'true'
+        Profile.objects.filter(user__username=username).update(research_ok=is_ok)
+        return HttpResponse()
+    if request.method == 'POST':  # Confirmed from mail
+        is_ok = 'yes' in request.POST
+        username = request.POST['username']
+        token = request.POST['token']
+    elif request.method == 'GET':  # Clicked on mail link
+        username = request.GET['username']
+        token = request.GET['token']
+    message = settings.HASH_NACL + username
+    if hashlib.sha1(message.encode('utf-8')).hexdigest() != token:  # If the token is invalid
```

Comment:
New in Python 3.3, awesome!

Toi je sens que tu vas faire des malheurs en crypto en master.